### PR TITLE
Accept address array in eth_newFilter params

### LINKED ISSUE #279: eth_newFilter implementation does not follow rpc spec allowing address to be an array
I got here by way of: https://github.com/ethereum/web3.py/issues/1019 and https://github.com/trufflesuite/ganache-core/issues/38

Support should be added to allow filters to match with multiple contracts, making the implementation spec compliant.

### PR DESCRIPTION
**Pull request by dylanjw**
Fixes #279

### CODE REVIEW

File: subproviders/filters.js [hunk 1]
```diff
@@ -345,7 +345,8 @@ function LogFilter(opts) {
   self.type = 'log'
   self.fromBlock = (opts.fromBlock !== undefined) ? opts.fromBlock : 'latest'
   self.toBlock = (opts.toBlock !== undefined) ? opts.toBlock : 'latest'
-  self.address = opts.address ? normalizeHex(opts.address) : opts.address
+  expectedAddress = opts.address && (Array.isArray(opts.address) ? opts.address : [opts.address]);
```
Comment (wjmelements) [inline subproviders/filters.js]: please declare this as a local variable

File: subproviders/filters.js [hunk 2]
```diff
@@ -362,7 +363,8 @@ LogFilter.prototype.validateLog = function(log){
 
   // address is correct:
   // console.log('LogFilter - validateLog - address', self.address)
-  if (self.address && self.address.toLowerCase() !== log.address.toLowerCase()) return false
+  if (self.address && !(self.address.map(
+    function(a) { return a.toLowerCase() }).includes(log.address.toLowerCase()))) return false
```
Comment (wjmelements) [inline subproviders/filters.js]: Use the shorthand function syntax:

```
self.address.map((a) => a.toLowerCase())
```

File: subproviders/filters.js (hunk 1 above)
Comment (wjmelements) [reply subproviders/filters.js]:
```
var expectedAddress = ....
```

File: subproviders/filters.js [hunk 3]
```diff
@@ -345,7 +345,8 @@ function LogFilter(opts) {
   self.type = 'log'
   self.fromBlock = (opts.fromBlock !== undefined) ? opts.fromBlock : 'latest'
   self.toBlock = (opts.toBlock !== undefined) ? opts.toBlock : 'latest'
-  self.address = opts.address ? normalizeHex(opts.address) : opts.address
+  expectedAddress = opts.address && (Array.isArray(opts.address) ? opts.address : [opts.address]);
+  self.address = expectedAddress && expectedAddress.map(function(a) { return normalizeHex(a) });
```
Comment (Revinand) [inline subproviders/filters.js]: this can be shortened as well

`self.address = expectedAddress && expectedAddress.map(normalizeHex)`

**Comment by dylanjw** [comment]
@wjmelements @Revinand Thanks!

**Comment by carver** [comment]
Who could give this PR the review that it needs for merging? Is that @Revinand ?

**Comment by Revinand** [comment]
Nope, it's not me. I'm not an admin nor have write permissions. Maybe @danfinlay could approve and merge this PR?

**Pull request merged**